app/src/main/java/com/example/sw0b_001/Homepage/GatewayClientListingFragment.kt

```diff
@@ -157,27 +176,28 @@ class GatewayClientListingFragment : Fragment(R.layout.activity_gateway_clients_
             view?.findViewById<MaterialTextView>(R.id.gateway_client_MSISDN)?.text =
                     gatewayClient.mSISDN
 
-            view?.setOnClickListener(gatewayClientOnClickListener(gatewayClient))
+            view?.setOnClickListener(gatewayClientOnClickListener(gatewayClient, viewModel))
 
             val radioButton = view?.findViewById<SwitchMaterial>(R.id.gateway_client_radio_btn)
             radioButton?.isChecked = defaultGatewayClientMsisdn == gatewayClient.mSISDN
-            radioButton?.setOnClickListener(gatewayClientOnClickListener(gatewayClient))
+            radioButton?.setOnClickListener(gatewayClientOnClickListener(gatewayClient, viewModel))
 
 
             view?.findViewById<MaterialCardView>(R.id.gateway_client_listing_card)
-                    ?.setOnClickListener(gatewayClientOnClickListener(gatewayClient))
+                    ?.setOnClickListener(gatewayClientOnClickListener(gatewayClient, viewModel))
 
             return view!!
         }
 
-        private fun gatewayClientOnClickListener(gatewayClient: GatewayClient):
+        private fun gatewayClientOnClickListener(gatewayClient: GatewayClient, viewModel: GatewayClientViewModel):
                 OnClickListener {
             return OnClickListener {
                 gatewayClientsCommunications.updateDefaultGatewayClient(gatewayClient.mSISDN!!)
                 CoroutineScope(Dispatchers.Default).launch{
                     gatewayClient.type = "custom"
                     Datastore.getDatastore(it.context).gatewayClientsDao().update(gatewayClient)
                 }
+                viewModel.updateSelectedGatewayClient(gatewayClient.mSISDN)
```

Comment:
You can call methods to update UI from here

Reply:
@sherlockwisdom I just made the changes